packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2431,13 +2439,60 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     updateKeepAlive();
   }
 
+  String _cachedText = '';
+  Rect? _cachedFirstRect;
+  Size _cachedSize = Size.zero;
+  int _cachedPlaceholder = -1;
+
+  void _updateSelectionRects({bool force = false}) {
+    if (defaultTargetPlatform != TargetPlatform.iOS)
+      return;
+    // This is to avoid sending selection rects on non-iPad devices
+    if (WidgetsBinding.instance!.window.physicalSize.shortestSide < 1536.0)
+      return;
+    final String text = buildTextSpan().toPlainText(includeSemanticsLabels: false, includePlaceholders: false);
+    final List<Rect> firstSelectionBoxes = renderEditable.getBoxesForSelection(const TextSelection(baseOffset: 0, extentOffset: 1));
+    final Rect? firstRect = firstSelectionBoxes.isNotEmpty ? firstSelectionBoxes.first : null;
+    final ScrollDirection scrollDirection = _scrollController?.position.userScrollDirection ?? ScrollDirection.idle;
+    final Size size = renderEditable.size;
+    if (scrollDirection == ScrollDirection.idle && (force || text != _cachedText ||
+        _cachedFirstRect != firstRect || _cachedSize != size ||
+        _cachedPlaceholder != _placeholderLocation)) {
```

Comment:
Would it make this any more readable if you split out some variable(s) out of this big conditional?